### ISSUE TITLE
DateTime scalar change in 1.10.11 broke backwards compatibility

### ISSUE THREAD

**Issue by denisahearn** [rmosolgo/graphql-ruby#2988]
**Describe the bug**

The changes introduced by pull request https://github.com/rmosolgo/graphql-ruby/pull/2973 broke backwards compatibility for this gem. 

If you have a mutation with a required input of type GraphQL::Types::ISO8601DateTime, and the input value provided to the mutation is a date such as 2020-06-12, versions 1.10.10 and earlier of this gem will accept this input and set the time element to 00:00:00. However version 1.10.11 will cause the mutation to fail and return an error like the following:

```
 {"message"=>"Argument 'reference_date' on Field 'create_shape' has an invalid value (\"2020-06-12\"). Expected type 'ISO8601DateTime!'.", "locations"=>[{"line"=>4, "column"=>5}], "path"=>["mutation", "create_shape", "reference_date"], "extensions"=>{"code"=>"argumentLiteralsIncompatible", "typeName"=>"Field", "argumentName"=>"reference_date"}}
```

You can see why this behavior has changed, by running this change made by the pull request in the Rails console:

```ruby
➜  rails c
> date_str = "2020-06-12"
 => "2020-06-12"
> Time.iso8601(date_str)
Traceback (most recent call last):
        1: from (irb):8
ArgumentError (invalid date: "2020-06-12")
> DateTime.iso8601(date_str)
 => Fri, 12 Jun 2020 00:00:00 +0000
```

Granted, this particular input should have been initially modeled using GraphQL::Types::ISO8601Date instead of GraphQL::Types::ISO8601DateTime, as it is stored as a date in our database. However, it wasn't, and making this change now would alter the API's schema and break any GraphQL client that is explicitly expecting this input to be of type GraphQL::Types::ISO8601DateTime.

**Versions**

`graphql` version: 1.10.11
`rails` (or other framework): We are using 6.0.3.1, but this issue is not specific to this Rails version
other applicable versions (`graphql-batch`, etc)

**GraphQL schema**

```
create_shape(
  name: String!
  reference_date: ISO8601DateTime!
  geometry: GeoJSON!
): Shape!
```

**GraphQL query**

```graphql
mutation CreateShape {
  create_shape(
    name: "Example Shape"
    reference_date: "2020-06-12"
    geometry: {
      type: "FeatureCollection",
      features: [
        {
          type: "Feature",
          properties: {},
          geometry: {
            type: "GeometryCollection",
            geometries: [
              {
                type: "Polygon",
                coordinates: [
                  [
                    [ -95.5948085, 47.0946385 ],
                    [ -95.5885835, 47.0947915 ],
                    [ -95.5893452, 47.0933998 ],
                    [ -95.5948085, 47.0946385 ]
                  ],
                  [
                    [ -95.591592, 47.0939598 ],
                    [ -95.5895655, 47.0945269 ],
                    [ -95.5908101, 47.0944966 ],
                    [ -95.591592, 47.0939598 ]
                  ]
                ]
              }
            ]
          }
        }
      ]
    }
  ) {
    sentera_id
    color
    acres
    reference_date
    geometry
  }
}
```

```json
{
  "errors": [
    {
      "message": "Argument 'reference_date' on Field 'create_shape' has an invalid value (\"2020-06-12\"). Expected type 'ISO8601DateTime!'.",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "mutation CreateShape",
        "create_shape",
        "reference_date"
      ],
      "extensions": {
        "code": "argumentLiteralsIncompatible",
        "typeName": "Field",
        "argumentName": "reference_date"
      }
    }
  ]
}
```

**Steps to reproduce**

Use the GraphQL schema and mutation listed above with version 1.10.11 of this gem to reproduce this bug.

**Expected behavior**

Passing a date formatted as ISO 8601 (ie. 2020-06-12) as a mutation input of type GraphQL::Types::ISO8601DateTime should work. The date should get converted with a 00:00:00 time element, which was the previous behavior of versions < 1.10.11).

**Actual behavior**

The mutation will fail as demonstrated by the mutation response JSON included above.

**Comment by rmosolgo** [comment]
Thanks for the detailed report! I agree it should not have broken compatibility.

**Comment by denisahearn** [comment]
@rmosolgo Thanks for responding. We'll pin to version 1.10.10 for now with the hopes that a fix for this issue will be released at some point.

**Comment by rmosolgo** [comment]
Thanks again for the report, I shipped a fix in 1.10.12. Please let me know if you run into more trouble!

**Comment by denisahearn** [comment]
@rmosolgo Version 1.11.0 fixes this issue for us. Thanks for the prompt response!
